core.js

```diff
@@ -911,13 +911,20 @@ async function _fromTokenizer(tokenizer) {
 		};
 	}
 
-	if (checkString('BEGIN:')) {
+	if (checkString('BEGIN:VCALEN')) { // Limit to 12 characters as precised in bytesRead const
```

Comment:
Just move it further down, where the sample length is increased.

Reply:
Not sure to really understand, do you want me I just move `BEGIN:VCARD` and `BEGIN:VCALEN` check under the `9-byte signatures` ?
Do you want I add a new `11-byte signature` comment too ?

Reply:
I recommend to move `BEGIN:VCARD` & `BEGIN:VCALENDER` to the appropriate locations.

Alternatively leave it here and keep `BEGIN:`
```js
if (checkString('BEGIN:')) { // So this is 6 byte test, the sub-tests take care of their own reads, so that is not taken into account
  await tokenizer.ignore(6); // advance pointer after 'BEGIN:'
  await tokenizer.readBuffer(buffer, 9, {mayBeLess: true}); // read word after BEGIN:'
  if (checkString('VCARD')) {
    return // vcard type
  } else if (checkString('VCALENDER')) {
    return // vcalander type
  }
  return; // as we permanently read, cannot proceed with other types
}

```

> Do you want I add a new 11-byte signature comment too ?

That is not necessary.